File: vcard.py
class VCardValue:
    def __init__(self, name, parameter, content):
        self.name = name
        self.parameter = parameter
        self.content = content
        
def parseLine(line):
    erg = line.split(':')
    content = erg[1]
    tmp = erg[0].split(';')
    name = erg[0]
    parameter = erg[1:]
    return VCardValue(name, parameter, content)
class VCard:
    def __init__(self, filename=""):
        if filename == "":
            self.name = ""
            self.icq = ""
            self.street = 0
            self.city = ""
        else:
            handle = open(filename)
            #while (line=handle.readline())==true:
            #    value=parseLine(line)
                

File: vcard.cpp
#include "vcard.h"
#include <QFile>
#include <iostream>
vCard::vCard(QStringList data)
{
    //QStringList lines =data.split('\n');
    std::cout<<"begin parsing"<<std::endl;
    foreach(QString line,data)
    {
        QStringList tmp = line.split(':');
        QStringList prop = tmp.at(0).split(';');
        if (prop.at(0) == "N")
            props[prop.at(0)] = new vCardName(prop.at(0),tmp.at(1));
        else
            props[prop.at(0)] = new vCardProperty(prop.at(0),tmp.at(1));
        //std::cout<<prop.at(0).toStdString()<<"::"<<tmp.at(1).toStdString()<<std::endl;

    }
}
vCardProperty* vCard::property(QString name)
{
    return props[name];
}

QList<vCard> vCard::readFromFile(QString filename)
{
    QList<vCard> vcards;
    QStringList tmp;
    QFile *card = new QFile(filename);
    if (!card->open(QIODevice::ReadOnly | QIODevice::Text))
        return vcards;

    while (!card->atEnd())
    {
        QString line =card->readLine();
        if(line.startsWith("END:VCARD") )
        {
            vcards.append(vCard(tmp));
            tmp.clear();
        }
        else if(!line.startsWith("BEGIN:VCARD"))
        {
            tmp.append(line);
        }


    }
    return vcards;
}
